src/web/app/components/dialogs/CodeGenerator/svgOperation.ts

```diff
@@ -0,0 +1,225 @@
+}
+
+/* Barcode */
+export async function importBarcodeSvgElement(
+  svgElement: SVGElement,
+  isInvert = false
+): Promise<void> {
+  const batchCmd = new history.BatchCommand('Import Barcode');
+  const doc = preProcessTextTag(svgElement);
+  const group = Array.of<SVGElement>();
+  const gElements = doc.querySelectorAll('g');
+  const fontObj = await getFontObj(
+    fontFuncs.getFontOfPostscriptName(doc.querySelector('text')?.getAttribute('font-postscript'))
+  );
+
+  gElements.forEach((g) => {
+    const transform = getTranslateValues(g.getAttribute('transform'));
+
+    g.querySelectorAll('rect').forEach((rect) => {
+      const { x, y, width, height } = rect.getBBox();
+
+      group.push(
+        svgCanvas.addSvgElementFromJson({
+          element: 'rect',
+          curStyles: false,
+          attr: {
+            x: x + transform.x,
+            y: y + transform.y,
+            width,
+            height,
+            stroke: '#000',
+            fill: 'black',
+            opacity: 1,
+            'fill-opacity': 1,
+            id: svgCanvas.getNextId(),
+          },
+        })
+      );
+    });
+
+    g.querySelectorAll('text').forEach((text) => {
+      const { textContent } = text;
+
+      if (!textContent) {
+        return;
+      }
+
+      const tspan = document.createElementNS(SVG_NS, 'tspan');
+
+      tspan.setAttribute('x', `${Number.parseFloat(text.getAttribute('x')) + transform.x}`);
+      tspan.setAttribute('y', `${Number.parseFloat(text.getAttribute('y')) + transform.y}`);
+      tspan.textContent = textContent;
+      text.textContent = '';
+      text.appendChild(tspan);
+
+      const { d } = convertTextToPathByFontkit(text, fontObj);
+
+      group.push(
+        svgCanvas.addSvgElementFromJson({
+          element: 'path',
+          curStyles: true,
+          attr: {
+            d,
+            fill: 'black',
+            opacity: 1,
+            'fill-opacity': 1,
+            id: svgCanvas.getNextId(),
+          },
+        })
+      );
+    });
+  });
+
+  group.forEach((element) => {
+    batchCmd.addSubCommand(new history.InsertElementCommand(element));
+  });
+
+  doc.remove();
+
+  svgCanvas.selectOnly(group);
+  batchCmd.addSubCommand(svgCanvas.groupSelectedElements(true));
+  svgCanvas.zoomSvgElem(10);
+
+  if (!batchCmd.isEmpty()) {
+    undoManager.addCommandToHistory(batchCmd);
+  }
+}
+
+/* QR Code */
+function handleQrCodeInvertColor(svgString: string, size: string): string {
+  const svgElement = document.createElementNS(SVG_NS, 'svg');
+
+  svgElement.setAttribute('xmlns', SVG_NS);
+  svgElement.setAttribute('height', '1000');
+  svgElement.setAttribute('width', '1000');
+  svgElement.setAttribute('viewBox', `0 0 ${size} ${size}`);
+
+  const fullBlackPath = document.createElementNS(SVG_NS, 'path');
+
+  fullBlackPath.setAttribute('fill', 'black');
+  fullBlackPath.setAttribute('d', `M0 0h${size}v${size}H0z`);
+
+  const codePathData = extractSvgTags(svgString, 'path')[1];
```

Comment:
Background rect is contained in original svg.
So we can fullBlackPathData from `extractSvgTags(svgString, 'path')`, too.